Add Apple Provisioning profile

georgehrke opened the pull request:
fixes #1434 

Todo:
- [x] create provisioning root collection and add apple provisioning node to it
- [x] tests

georgehrke committed:
Add Apple Provisioning profile

Signed-off-by: Georg Ehrke <developer@georgehrke.com>

georgehrke commented:
It seems that even with the most detailed provisioning profile iOS / macOS will still look for the .well-known/caldav  / .well-known/carddav file :/ 

https://netfuture.ch/2013/07/dealing-with-apple-mobileconfig-configuration-profiles-quirks/

@MorrisJobke @rullzer Did we fix that issue that .well-known is only installed inside the Nextcloud-folder but not in the domains root?

MorrisJobke commented:
We did not, because it then needs to be configured in the Webserver itself

jospoortvliet commented:
So that can't be 100% fixed? :sob:

MorrisJobke commented:
> @MorrisJobke @rullzer Did we fix that issue that .well-known is only installed inside the Nextcloud-folder but not in the domains root?

We have a setup check for this nowadays. So I would assume that it will work for now and provide it.

MorrisJobke commented:
I rebased it. I would say, we should continue this and can improve in the future iteratively.

MorrisJobke commented:
@georgehrke Looks like this one will not make it in time, right? Would like to have it merged by tomorrow evening.

rullzer commented:
Lets rebase and get this in. Yeah it won't work in 100% of the cases but so be it there is no way to do that.

georgehrke commented:
We should remove `CalDAVPrincipalURL` and `CardDAVPrincipalURL`, because they are not used anymore. 

Will rebase, fix that and add tests next week.

MorrisJobke commented:
> Will rebase, fix that and add tests next week.

Could you have a look at this? So we can finally merge it :)

georgehrke commented:
tl;dr of what changed:
- [x] added unit tests
- [x] enforce HTTPs. Using the profile on HTTP silently fails.
- [x] removed principalURI and require admins to properly configure well-known.

georgehrke commented:
please review @nextcloud/mac

rullzer commented:
I can't test since.. well no mac...

@georgehrke for a followup PR what would be nice

1. Add a check for the well-known
2. On the groupware page detect if a user is on a Apple device and show a link

rullzer reviewed (approved):
Looks good from my side. And tests :tada: 
But I can't tests since no macOS/iOS. so. :elephant:

marinofaggiana commented:
> Looks good from my side. And tests 🎉
> But I can't tests since no macOS/iOS. so. 🐘

At end of next week I prepare a test for iOS

cc @georgehrke

MorrisJobke commented on apps/dav/lib/Provisioning/Apple/AppleProvisioningPlugin.php:
```diff
@@ -0,0 +1,269 @@
+	 * @param \closure $uuidClosure
+	 */
+	public function __construct(IUserSession $userSession, IURLGenerator $urlGenerator,
+								ThemingDefaults $themingDefaults, IRequest $request,
```
This will fail if the theming app is disabled. Let me check if I can fix this.

MorrisJobke committed:
Fix theming defaults injection if theming app is disabled

Signed-off-by: Morris Jobke <hey@morrisjobke.de>

MorrisJobke reviewed (approved):
Tested and works really nice 👍
